tests/Functional/Parameters/ParameterTest.php

```diff
@@ -121,18 +121,83 @@ public function testHeaderParameterRequired(): void
+        yield 'valid boolean 1 string header_boolean' => ['header_boolean', ['Lorem' => '1'], 200];
+        yield 'invalid boolean header_boolean' => ['header_boolean', ['Lorem' => 'string'], 422];
+
+        // header_null
+        yield 'missing header header_null' => ['header_null', [], 200];
+        yield 'valid null header_null' => ['header_null', ['Ipsum' => null], 200];
+        yield 'valid null string header_null' => ['header_null', ['Ipsum' => 'null'], 200];
+        yield 'invalid null header_null' => ['header_null', ['Ipsum' => 'string'], 422];
+    }
+
+    #[DataProvider('provideQueryValues')]
+    public function testQueryParameter(string $url, array $query, int $expectedStatusCode): void
+    {
+        self::createClient()->request('GET', $url, ['query' => $query]);
+        $this->assertResponseStatusCodeSame($expectedStatusCode);
+    }
+
+    public static function provideQueryValues(): iterable
+    {
+        // query_integer
+        yield 'valid integer query_integer' => ['query_integer', ['Foo' => '3'], 200];
+        yield 'too high query_integer' => ['query_integer', ['Foo' => '6'], 422];
+        yield 'too low query_integer' => ['query_integer', ['Foo' => '0'], 422];
+        yield 'invalid integer query_integer' => ['query_integer', ['Foo' => 'string'], 422];
+
+        // query_float
+        yield 'valid float query_float' => ['query_float', ['Bar' => '3.5'], 200];
+        yield 'valid integer query_float' => ['query_float', ['Bar' => '3'], 200];
+        yield 'too high query_float' => ['query_float', ['Bar' => '600'], 422];
+        yield 'too low query_float' => ['query_float', ['Bar' => '0'], 422];
+        yield 'invalid number query_float' => ['query_float', ['Bar' => 'string'], 422];
+
+        // query_boolean
+        yield 'valid boolean false query_boolean' => ['query_boolean', ['Lorem' => false], 200];
+        yield 'valid boolean false string query_boolean' => ['query_boolean', ['Lorem' => 'false'], 200];
+        yield 'valid boolean true query_boolean' => ['query_boolean', ['Lorem' => true], 200];
+        yield 'valid boolean true string query_boolean' => ['query_boolean', ['Lorem' => 'true'], 200];
+        yield 'valid boolean 0 query_boolean' => ['query_boolean', ['Lorem' => 0], 200];
+        yield 'valid boolean 0 string query_boolean' => ['query_boolean', ['Lorem' => '0'], 200];
+        yield 'valid boolean 1 query_boolean' => ['query_boolean', ['Lorem' => 1], 200];
+        yield 'valid boolean 1 string query_boolean' => ['query_boolean', ['Lorem' => '1'], 200];
+        yield 'invalid boolean query_boolean' => ['query_boolean', ['Lorem' => 'string'], 422];
+
+        // query_null
+        yield 'valid null query_null' => ['query_null', ['Ipsum' => null], 200];
+        yield 'valid null string query_null' => ['query_null', ['Ipsum' => 'null'], 200];
+        yield 'invalid null query_null' => ['query_null', ['Ipsum' => 'string'], 422];
```

Comment:
I often use things like this: `my_route?foo` and actually test the presence of the parameter (value is ParameterNotFound), what would be the behavior in your opinion is my parameter `null` or an empty string? I do agree that it's nice to support it as its a JSON Schema type but I don't see any usage for parameters.